terraform/modules/database/rds.tf

```
@@ -2,8 +2,24 @@ resource "aws_db_subnet_group" "cluster" {
   subnet_ids = var.vpc.aws_subnets_private[*].id
 }
 
+resource "aws_kms_key" "aurora" {
+  enable_key_rotation     = true
+  deletion_window_in_days = 7
+  tags                    = merge(var.tags, { Name = "${var.name}-db-key", ProtectsSensitiveData = true })
+}
+
+resource "aws_kms_alias" "aurora" {
+  name          = "alias/${var.name}-db-key"
+  target_key_id = aws_kms_key.aurora.key_id
+}
+
+data "aws_db_cluster_snapshot" "cluster" {
+  db_cluster_snapshot_identifier = var.name
```

might be a be better coupled to the resource
```suggestion
  db_cluster_snapshot_identifier = aws_rds_cluster.cluster.cluster_identifier
```

---

this would be a cyclical dependencies, so ignore me